container-entrypoint.sh

```diff
@@ -1,7 +1,68 @@
 #!/bin/bash
 
-# in case of using cache dir, we need to initialize it
-/usr/sbin/squid -d 1 --foreground -f /etc/squid/squid.conf -z
+set -euo pipefail
 
-# now start the squid primary process with supplied options
-/usr/sbin/squid -d 1 --foreground -f /etc/squid/squid.conf $@
+# Configuration from environment variables
+ENABLE_PER_SITE_EXPORTER="${ENABLE_PER_SITE_EXPORTER:-false}"
+# Note: WEB_LISTEN_ADDRESS, WEB_TLS_CERT_FILE, WEB_TLS_KEY_FILE are read directly by the Go application
+
+start_per_site_exporter() {
+    # Configuration is handled via environment variables (see main.go)
+    # The Go application reads WEB_LISTEN_ADDRESS, WEB_TLS_CERT_FILE, WEB_TLS_KEY_FILE
+    local listen_addr="${WEB_LISTEN_ADDRESS:-:9302}"
+    
+    if [[ -n "${WEB_TLS_CERT_FILE:-}" && -n "${WEB_TLS_KEY_FILE:-}" ]]; then
+        echo "Starting per-site exporter with TLS on $listen_addr"
+    else
+        echo "Starting per-site exporter on $listen_addr"
+    fi
+    
+    # No need to pass flags - the Go app reads environment variables directly
+    exec /usr/local/bin/squid-per-site-exporter
+}
+
+start_squid() {
+    echo "Initializing squid cache directory..."
+    /usr/sbin/squid -f /etc/squid/squid.conf -z
+    
+    # Remove PID file created by -z initialization to avoid conflicts
+    rm -f /run/squid/squid.pid
+
+    echo "Starting squid proxy..."
+    # Note: access_log goes to stdout (configured in squid.conf), cache_log goes to stderr
+    # Remove -d flag to avoid mixing debug output with access logs
+    # Pass all script arguments to squid
+    exec /usr/sbin/squid --foreground -f /etc/squid/squid.conf "$@"
+}
+
+# Main execution logic
+if [[ "$ENABLE_PER_SITE_EXPORTER" == "true" ]]; then
+    echo "Starting squid with per-site metrics exporter..."
+    
+    # Create a named pipe for log streaming
+    LOG_PIPE="/tmp/squid-access-logs"
```

Comment:
why do we need all this complex logic and a named pipe instead of doing this:
```
start_squid "@$" | start_per_site_exporter
```

Then we only need to make sure the exporter prints back its output so that we also get in in STDOUT